components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java

```diff
@@ -1814,13 +1816,18 @@ client createSession(String userName, String password, String hostName,
 	{
 		this.encrypted = encrypted;
 		client secureClient = null;
+		
 		try {
 		    // client must be cleaned up by caller.
 			if (port > 0) secureClient = new client(hostName, port);
 			else secureClient = new client(hostName);
 			secureClient.setAgent(agentName);
-			ServiceFactoryPrx entryEncrypted
-			    = secureClient.createSession(userName, password);
+			ServiceFactoryPrx entryEncrypted;
+			if (userName.equals(password)) {
```

Comment:
`joinSession(userName)` == `createSession(userName, userName)` so this should be fine, but you could try using the guest connection to look for a user of this name if you want to be really careful.